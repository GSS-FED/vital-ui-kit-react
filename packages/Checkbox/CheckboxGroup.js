// @flow

import * as React from 'react';
import styled from 'styled-components';

const StyledCheckboxGroup = styled.div`
`;

type Props = {
  children: React.Node,
};

const CheckboxGroup = ({ children, ...props }: Props) => (
  <StyledCheckboxGroup {...props}>{children}</StyledCheckboxGroup>
);

export default CheckboxGroup;
