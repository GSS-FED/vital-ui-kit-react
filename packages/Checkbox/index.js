export { default as Checkbox } from './Checkbox';
export { default as CheckboxGroup } from './CheckboxGroup';

/** @render react
 * @name CheckboxGroup
 * @description Group of checkboxes
 * @example
 * <CheckboxGroup>
 *  <Checkbox
 *    isRound
 *    label="checkbox 1"
 *  />
 *  <Checkbox
 *    isRound
 *    label="checkbox 2"
 *  />
 *  <Checkbox
 *    isRound
 *    label="checkbox 3"
 *  />
 * </CheckboxGroup>
 */
