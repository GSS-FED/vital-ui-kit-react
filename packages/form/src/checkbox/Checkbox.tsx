/**
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled from 'styled-components';
import { rgba } from 'polished';
import { CheckIcon } from './CheckIcon';
import { defaultTheme } from '@vital-ui/react-theme';
import cn from 'classnames';
import { CheckboxContext } from './CheckboxContext';

const Root = styled.label`
  font-size: 15px;
  cursor: pointer;
  margin-right: 5px;
  &:hover {
    > span:first-child {
      box-shadow: ${({ theme }) =>
        `0 0 0 2px ${rgba(theme.colors.primary, 0.2)}`};
    }
  }
`;

Root.defaultProps = {
  theme: defaultTheme,
};

const Label = styled.span`
  color: ${({ theme }) => theme.checkbox.label.color};
`;

Label.defaultProps = {
  theme: defaultTheme,
};

interface CheckWrapper {
  checked?: boolean;
  round?: boolean;
}

const CheckWrapper = styled('span')<CheckWrapper>`
  box-sizing: border-box;
  position: relative;
  display: inline-block;
  width: 1.066rem;
  height: 1.066rem;
  border: ${({ theme }) => `1px solid ${theme.checkbox.borderColor}`};
  border-radius: ${props => (props.round ? '50%' : '2px')};
  background-color: ${({ checked, round, theme }) =>
    checked
      ? theme.checkbox.checkedBg
      : round
      ? theme.checkbox.roundBg
      : theme.checkbox.bg};
  transition: ${({ theme }) => theme.defaultTransition};
  margin: -2px 0.6em 0 0;
  vertical-align: middle;

  border-color: ${({ checked, theme }) =>
    checked ? theme.checkbox.checkedBorderColor : ''};
`;

CheckWrapper.defaultProps = {
  theme: defaultTheme,
};

const Input = styled.input`
  display: none;
`;

interface IconWrapperProps {
  checked?: boolean;
  round?: boolean;
  disabled?: boolean;
}

const IconWrapper = styled('div')<IconWrapperProps>`
  text-align: center;
  width: 100%;
  height: 100%;
  position: relative;
  opacity: ${({ checked }) => (checked ? 1 : 0)};
  transform: ${({ checked }) => (checked ? 'scale(1)' : 'scale(0)')};
  pointer-events: none;
  transition: all 120ms ease-out;
  color: ${({ round, disabled, theme }) =>
    iconColor(round, disabled, theme)};
`;

IconWrapper.defaultProps = {
  theme: defaultTheme,
};

export interface CheckboxProps {
  /** Boolean checked value of the checkbox */
  checked?: boolean;
  /** Inital Check value */
  defaultChecked?: boolean;
  /** Disabled checkbox */
  disabled?: boolean;
  /** Round style */
  round?: boolean;
  /** Custom checkbox Icon */
  icon?: React.ReactNode | ((props: any) => React.ReactNode);
  /** Label text after the checkbox */
  label?: string;
  /** Html name attribute */
  name?: string;
  /** Function trigger when checkbox value changes */
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onValueChange?: (value: boolean) => void;
  /** Theme */
  theme?: any;
  style?: React.CSSProperties;
  /** default: `vital__checkbox` */
  className?: string;
}

function iconColor(
  round?: boolean,
  disabled?: boolean,
  theme: any = defaultTheme,
) {
  if (disabled) {
    return theme.checkbox.icon.disabledColor;
  }
  if (round) {
    return theme.checkbox.icon.roundColor;
  }
  return theme.checkbox.icon.color;
}

/**
 * @render react
 * @name Checkbox
 * @description Checkbox components with group
 * @example
 * <Checkbox.Group>
 *  <Checkbox
 *    round
 *    label="checkbox 1"
 *  />
 *  <Checkbox
 *    round
 *    label="checkbox 2"
 *  />
 *  <Checkbox
 *    round
 *    label="checkbox 3"
 *  />
 * </Checkbox.Group>
 */
export class Checkbox extends React.Component<CheckboxProps> {
  static defaultProps = {
    disabled: false,
    round: false,
    icon: undefined,
    theme: defaultTheme,
    style: undefined,
    className: '',
  };

  render() {
    const {
      defaultChecked,
      disabled = false,
      icon,
      label,
      name,
      round = false,
      theme,
      style,
      className,
      onChange,
      onValueChange,
      ...props
    } = this.props;

    return (
      <CheckboxContext.Consumer>
        {({
          round: contextRound,
          disabled: contextDisabled,
          icon: contextIcon,
        }) => (
          <Root
            style={style}
            className={cn('vital__checkbox', className)}
            {...props}
          >
            <CheckWrapper
              checked={this.props.checked}
              round={contextRound || round}
            >
              <IconWrapper checked={this.props.checked}>
                {contextIcon || icon || <CheckIcon />}
              </IconWrapper>
            </CheckWrapper>
            <Label>
              <Input
                type="checkbox"
                disabled={contextDisabled || disabled}
                checked={this.props.checked}
                defaultChecked={defaultChecked}
                name={name}
                onChange={e => {
                  if (onChange) {
                    onChange(e);
                  }
                  if (onValueChange) {
                    onValueChange(!this.props.checked);
                  }
                }}
              />
              {label}
            </Label>
          </Root>
        )}
      </CheckboxContext.Consumer>
    );
  }
}
