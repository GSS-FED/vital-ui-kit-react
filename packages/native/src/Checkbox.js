// @flow
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import Icon from 'react-native-vector-icons/Ionicons';
import boxShadow from './helper/boxShadow';

const Root = styled.View`
  flex-direction: row;
  align-items: center;
`;

const Box = styled.View`
  width: ${props => props.size || 16};
  height: ${props => props.size || 16};
  border-radius: 3;
  background: white;
  margin-right: 17;
  ${boxShadow(0, 4, 5, '#000', 0.17)};
`;

const Check = styled(Icon)`
  position: absolute;
  left: 2;
`;

export type CheckboxProps = {
  checked: boolean,
  size: number,
  onChange?: (checked: boolean) => {},
};

export type CheckboxState = {
  checked: boolean,
};

class Checkbox extends React.Component<CheckboxProps, CheckboxState> {
  static defaultProps = {
    checked: false,
    size: 16,
  };
  state = {
    checked: this.props.checked,
  };

  componentWillReceiveProps(nextProps: P) {
    this.setState({
      checked: nextProps.checked,
    });
  }

  onToggle = () => {
    this.setState({
      checked: !this.state.checked,
    });
    if (this.props.onChange) {
      this.props.onChange(this.state.checked);
    }
  };

  render() {
    return (
      <TouchableOpacity onPress={this.onToggle}>
        <Root checked={this.state.checked}>
          <Box size={this.props.size}>
            {this.state.checked && (
              <Check
                color="#999"
                name="md-checkmark"
                size={this.props.size}
              />
            )}
          </Box>
        </Root>
      </TouchableOpacity>
    );
  }
}

export default Checkbox;
