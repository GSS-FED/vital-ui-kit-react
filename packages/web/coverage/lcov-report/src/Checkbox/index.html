<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Checkbox</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> src/Checkbox
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">70.97% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/28</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Checkbox.jsx"><a href="Checkbox.jsx.html">Checkbox.jsx</a></td>
	<td data-value="67.86" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 67%;"></div><div class="cover-empty" style="width:33%;"></div></div></td>
	<td data-value="67.86" class="pct medium">67.86%</td>
	<td data-value="28" class="abs medium">19/28</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="76" class="pct medium">76%</td>
	<td data-value="25" class="abs medium">19/25</td>
	</tr>

<tr>
	<td class="file high" data-value="CheckboxGroup.jsx"><a href="CheckboxGroup.jsx.html">CheckboxGroup.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 05 2018 19:42:58 GMT+0800 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
