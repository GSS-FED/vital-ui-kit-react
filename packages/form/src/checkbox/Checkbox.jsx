/**
 * @flow
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled, { withTheme } from 'styled-components';
import { rgba } from 'polished';
import Icon from '@vital-ui/react-icon';
import { defaultTheme } from '@vital-ui/react-theme';
import cn from 'classnames';

import CheckboxGroup from './CheckboxGroup';

const Root = styled.label`
  font-size: 15px;
  cursor: pointer;
  margin-right: 5px;
  &:hover {
    > span:first-child {
      box-shadow: ${({ theme }) =>
        `0 0 0 2px ${rgba(theme.primary, 0.2)}`};
    }
  }
`;
const Label = styled.span`
  color: ${({ theme }) => theme.checkbox.label.color};
`;

const Box = styled.span`
  box-sizing: border-box;
  position: relative;
  display: inline-block;
  width: 1.066rem;
  height: 1.066rem;
  border: ${({ theme }) => `1px solid ${theme.checkbox.borderColor}`};
  border-radius: ${props => (props.round ? '50%' : '2px')};
  background-color: ${({ checked, round, theme }) =>
    theme.checkbox.bg(checked && round)};
  transition: ${({ theme }) => theme.defaultTransition};
  margin: -2px 0.6em 0 0;
  vertical-align: middle;

  border-color: ${({ checked, theme }) =>
    checked ? theme.checkbox.checkedBorderColor : ''};
`;

const Input = styled.input`
  display: none;
`;

const IconWrapper = styled.div`
  text-align: center;
  width: 100%;
  height: 100%;
  position: relative;
  opacity: ${({ checked }) => (checked ? 1 : 0)};
  transform: ${({ checked }) => (checked ? 'scale(1)' : 'scale(0)')};
  pointer-events: none;
  transition: all 120ms ease-out;
`;

type State = {
  checked: boolean,
};

export type Props = {
  /** Boolean checked value of the checkbox */
  checked: boolean,
  /** Inital Check value */
  defaultChecked?: boolean,
  /** Disabled checkbox */
  disabled?: boolean,
  /** Round style */
  round?: boolean,
  /** Custom checkbox Icon */
  icon?: React.Node,
  /** Label text after the checkbox */
  label: string,
  /** Html name attribute */
  name: string,
  /** Function trigger when checkbox value changes */
  onChange: (props: any) => {},
  /** Theme */
  theme?: Object,
  /** Value of the checkbox, html value attribute */
  value: number | string,
  style?: CSSStyleDeclaration,
  /** default: `vital__checkbox` */
  className?: string,
};

function iconColor(round, disabled, theme = defaultTheme) {
  if (disabled) {
    return theme.checkbox.icon.disabledColor;
  }
  if (round) {
    return theme.checkbox.icon.roundColor;
  }
  return theme.checkbox.icon.color;
}

/** @render react
 * @name Checkbox
 * @description Checkbox components with group
 * @example
 * <Checkbox.Group>
 *  <Checkbox
 *    round
 *    label="checkbox 1"
 *  />
 *  <Checkbox
 *    round
 *    label="checkbox 2"
 *  />
 *  <Checkbox
 *    round
 *    label="checkbox 3"
 *  />
 * </Checkbox.Group>
 */

class Checkbox extends React.Component<Props, State> {
  static defaultProps = {
    defaultChecked: false,
    disabled: false,
    round: false,
    icon: undefined,
    theme: defaultTheme,
    style: undefined,
    className: '',
  };

  static Group = CheckboxGroup;

  state = {
    checked:
      'checked' in this.props
        ? this.props.checked
        : this.props.defaultChecked || false,
  };

  static getDerivedStateFromProps(props: Props) {
    if ('checked' in props) {
      return {
        checked: props.checked,
      };
    }
    return null;
  }

  handleChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    const { disabled, ...props } = this.props;
    if (disabled) return;
    this.props.onChange({
      target: {
        ...props,
        checked: e.target.checked,
      },
      preventDefault() {
        e.preventDefault();
      },
      nativeEvent: e.nativeEvent,
    });
  };

  render() {
    const {
      defaultChecked,
      disabled,
      icon,
      label,
      name,
      round,
      theme,
      value,
      style,
      className,
      ...props
    } = this.props;

    const customIcon = icon && typeof icon !== 'string';

    return (
      <Root
        style={style}
        className={cn('vital__checkbox', className)}
        {...props}
      >
        <Box
          checked={this.state.checked}
          disabled={disabled}
          round={round}
        >
          <IconWrapper
            checked={this.state.checked}
            disabled={disabled}
          >
            {customIcon || (
              <Icon
                name={icon || 'check'}
                size="12"
                color={iconColor(round, disabled, theme)}
              />
            )}
          </IconWrapper>
        </Box>
        <Label checked={this.state.checked} disabled={disabled}>
          <Input
            type="checkbox"
            disabled={disabled}
            checked={this.state.checked}
            defaultChecked={defaultChecked}
            name={name}
            value={value}
            onChange={() => this.handleChange}
          />
          {label}
        </Label>
      </Root>
    );
  }
}

export default withTheme(Checkbox);
