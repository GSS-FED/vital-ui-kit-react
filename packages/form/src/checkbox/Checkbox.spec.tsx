import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from '@vital-ui/react-theme';
import { Checkbox } from './Checkbox';

describe('Checkbox', () => {
  it('should render with default styles', () => {
    const checkbox = render(<Checkbox label="default" checked />);
    expect(checkbox).toMatchSnapshot();
  });

  it('should render with Custom Theme', () => {
    const checkbox = render(
      <ThemeProvider
        componentTheme={props => ({
          checkbox: {
            label: {
              color: 'red',
            },
          },
        })}
      >
        <Checkbox label="custom" checked />
      </ThemeProvider>,
    );
    expect(checkbox).toMatchSnapshot();
  });

  it('render with round style', () => {
    const checkbox = render(<Checkbox label="round" round />);
    expect(checkbox).toMatchSnapshot();
  });
});
