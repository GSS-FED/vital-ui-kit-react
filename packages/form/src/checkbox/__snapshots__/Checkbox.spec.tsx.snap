// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render with round style 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  font-size: 15px;
  cursor: pointer;
  margin-right: 5px;
  display: inline-block;
}

.c0:hover > span:first-child {
  box-shadow: 0 0 0 2px rgba(14,134,254,0.2);
}

.c3 {
  color: #456296;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: inline-block;
  width: 1.066rem;
  height: 1.066rem;
  border: 1px solid #C3D1EB;
  border-radius: 50%;
  background-color: #ffffff;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  margin: -2px 0.6em 0 0;
  vertical-align: middle;
}

.c4 {
  display: none;
}

.c2 {
  text-align: center;
  width: 100%;
  height: 100%;
  position: relative;
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  pointer-events: none;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  color: #0E86FE;
}

<div>
      <label
        class="vital__checkbox c0"
      >
        <span
          class="c1"
        >
          <div
            class="c2"
          >
            <svg
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </div>
        </span>
        <span
          class="c3"
        >
          <input
            class="c4"
            type="checkbox"
          />
          round
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="vital__checkbox sc-bdVaJa ddktzo"
    >
      <span
        class="sc-htpNat cUzqxB"
      >
        <div
          class="sc-ifAKCX hLQgOo"
        >
          <svg
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </div>
      </span>
      <span
        class="sc-bwzfXH iuQrxO"
      >
        <input
          class="sc-bxivhb cRvGhf"
          type="checkbox"
        />
        round
      </span>
    </label>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox should render with Custom Theme 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  font-size: 15px;
  cursor: pointer;
  margin-right: 5px;
  display: inline-block;
}

.c0:hover > span:first-child {
  box-shadow: 0 0 0 2px rgba(14,134,254,0.2);
}

.c3 {
  color: red;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: inline-block;
  width: 1.066rem;
  height: 1.066rem;
  border: 1px solid #C3D1EB;
  border-radius: 2px;
  background-color: #ffffff;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  margin: -2px 0.6em 0 0;
  vertical-align: middle;
  border-color: #95ABD3;
}

.c4 {
  display: none;
}

.c2 {
  text-align: center;
  width: 100%;
  height: 100%;
  position: relative;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  pointer-events: none;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  color: #0E86FE;
}

<div>
      <label
        class="vital__checkbox c0"
      >
        <span
          class="c1"
        >
          <div
            class="c2"
          >
            <svg
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </div>
        </span>
        <span
          class="c3"
        >
          <input
            checked=""
            class="c4"
            type="checkbox"
          />
          custom
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="vital__checkbox sc-bdVaJa ddktzo"
    >
      <span
        class="sc-htpNat deAzeE"
      >
        <div
          class="sc-ifAKCX jXwSsA"
        >
          <svg
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </div>
      </span>
      <span
        class="sc-bwzfXH bukUYK"
      >
        <input
          checked=""
          class="sc-bxivhb cRvGhf"
          type="checkbox"
        />
        custom
      </span>
    </label>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox should render with default styles 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  font-size: 15px;
  cursor: pointer;
  margin-right: 5px;
  display: inline-block;
}

.c0:hover > span:first-child {
  box-shadow: 0 0 0 2px rgba(14,134,254,0.2);
}

.c3 {
  color: #456296;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: inline-block;
  width: 1.066rem;
  height: 1.066rem;
  border: 1px solid #C3D1EB;
  border-radius: 2px;
  background-color: #ffffff;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  margin: -2px 0.6em 0 0;
  vertical-align: middle;
  border-color: #95ABD3;
}

.c4 {
  display: none;
}

.c2 {
  text-align: center;
  width: 100%;
  height: 100%;
  position: relative;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  pointer-events: none;
  -webkit-transition: all 120ms ease-out;
  transition: all 120ms ease-out;
  color: #0E86FE;
}

<body>
    <div>
      <label
        class="vital__checkbox c0"
      >
        <span
          class="c1"
        >
          <div
            class="c2"
          >
            <svg
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </div>
        </span>
        <span
          class="c3"
        >
          <input
            checked=""
            class="c4"
            type="checkbox"
          />
          default
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="vital__checkbox sc-bdVaJa ddktzo"
    >
      <span
        class="sc-htpNat deAzeE"
      >
        <div
          class="sc-ifAKCX jXwSsA"
        >
          <svg
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </div>
      </span>
      <span
        class="sc-bwzfXH iuQrxO"
      >
        <input
          checked=""
          class="sc-bxivhb cRvGhf"
          type="checkbox"
        />
        default
      </span>
    </label>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
